File: src/main/java/cc/duduhuo/git/conflict/model/ConflictItem.java
package cc.duduhuo.git.conflict.model;

/**
 * =======================================================
 * Author: liying - liruoer2008@yeah.net
 * Datetime: 2018/4/21 19:03
 * Description:
 * Remarks:
 * =======================================================
 */
public class ConflictItem {
    private int currentChangeLineNum = -1;
    private int separatorLineNum = -1;
    private int incomingLineNum = -1;
    private String currentChangeStr;
    private String incomingChangeStr;

    public ConflictItem() {
    }

    public ConflictItem(int currentChangeLineNum, int separatorLineNum, int incomingLineNum) {
        this.currentChangeLineNum = currentChangeLineNum;
        this.separatorLineNum = separatorLineNum;
        this.incomingLineNum = incomingLineNum;
    }

    public int getCurrentChangeLineNum() {
        return currentChangeLineNum;
    }

    public void setCurrentChangeLineNum(int currentChangeLineNum) {
        this.currentChangeLineNum = currentChangeLineNum;
    }

    public int getSeparatorLineNum() {
        return separatorLineNum;
    }

    public void setSeparatorLineNum(int separatorLineNum) {
        this.separatorLineNum = separatorLineNum;
    }

    public int getIncomingLineNum() {
        return incomingLineNum;
    }

    public void setIncomingLineNum(int incomingLineNum) {
        this.incomingLineNum = incomingLineNum;
    }

    public String getCurrentChangeStr() {
        return currentChangeStr;
    }

    public void setCurrentChangeStr(String currentChangeStr) {
        this.currentChangeStr = currentChangeStr;
    }

    public String getIncomingChangeStr() {
        return incomingChangeStr;
    }

    public void setIncomingChangeStr(String incomingChangeStr) {
        this.incomingChangeStr = incomingChangeStr;
    }

    @Override
    public String toString() {
        return "ConflictItem{" +
            "currentChangeLineNum=" + currentChangeLineNum +
            ", separatorLineNum=" + separatorLineNum +
            ", incomingLineNum=" + incomingLineNum +
            ", currentChangeStr='" + currentChangeStr + '\'' +
            ", incomingChangeStr='" + incomingChangeStr + '\'' +
            '}';
    }
}

File: src/main/java/cc/duduhuo/git/conflict/model/ConflictItem.kt
package cc.duduhuo.git.conflict.model

/**
 * =======================================================
 * Author: liying - liruoer2008@yeah.net
 * Datetime: 2018/4/21 19:03
 * Description: 冲突条目对象
 * Remarks: StartLineNum 包含，EndLineNum 不包含
 * =======================================================
 */
data class ConflictItem(
    var currentChangeMarkerLineNum: Int = -1,
    var currentChangeStartLineNum: Int = -1,
    var currentChangeEndLineNum: Int = -1,

    var commonMarkerLineNum: Int = -1,
    var commonStartLineNum: Int = -1,
    var commonEndLineNum: Int = -1,

    var separatorMarkerLineNum: Int = -1,

    var incomingChangeMarkerLineNum: Int = -1,
    var incomingChangeStartLineNum: Int = -1,
    var incomingChangeEndLineNum: Int = -1,

    var currentChangeStr: String = "",
    var commonStr: String = "",
    var incomingChangeStr: String = "",
)
